.github/workflows/spotbugs.yml

```diff
@@ -0,0 +1,136 @@
+name: SpotBugs
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  spotbugs:
+    name: Analyze with SpotBugs
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      pull-requests: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Setup Java
+        uses: ./.github/actions/setup-java
+
+      - name: Build the project (without tests)
+        run: ./mvnw install -Pdistribution -DskipTests
+
+      - name: Resolve dependencies
+        run: ./mvnw dependency:resolve -V -e -DskipTests -DexcludeGroupIds=org.infinispan -Dsilent=true
+
+      - name: Get changed Java files
+        id: changed-files
+        run: |
+            echo "CHANGED_FILES=$(git diff --name-only origin/main...HEAD -- '*.java' | tr '\n' ' ')" >> $GITHUB_ENV
```

Comment:
this fails on a PR: see [Get Changed Java files  in this run](https://github.com/rigazilla/infinispan/actions/runs/13809757530/job/38628455299?pr=76).
I think the target branch needs to be fetched, something like: `git fetch origin <target branch> --depth=1`

also some PRs aren't created against main, but on other branches, so we may want to use a github variable like `GITHUB_BASE_REF`